### PR TITLE
Journal Entry Export

### PR DESCRIPTION
**Pull request by ruuushhh**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new search functionality for accounting exports, errors, and summaries.
  - Added ability to update accounting export summaries with counts of failed and successful exports.
  - Implemented a new `AccountingDataExporter` class for exporting data to external accounting systems.
  - Added new models for journal entries, line items, and purchase invoices with creation and update capabilities.
  - Added a new URL path for triggering exports within a workspace.
  - Enhanced expense management with the introduction of `ExpenseFilter` and `Expense` models.

- **Improvements**
  - Updated `TYPE_CHOICES` to reflect more accurate accounting export types.
  - Improved error and exception handling for business central operations.
  - Enhanced the export process with new functions and classes for journal entries and purchase invoices.
  - Added new utility methods for better integration with Fyle and Business Central.
  - Improved workspace model with additional fields and updated export mode choices.
  - Updated `requirements.txt` with newer versions of dependencies for better performance and stability.

- **Bug Fixes**
  - Fixed an issue with expense object creation to ensure proper association with accounting exports.

- **Documentation**
  - Updated documentation to reflect new features and improvements in accounting exports and workspace management.

- **Refactor**
  - Reorganized code related to expense import and synchronization for better maintainability.
  - Cleaned up import statements and removed unused imports across various files.

- **Style**
  - Modified code styling for consistency and readability in models and utility files.

- **Chores**
  - Performed routine codebase maintenance tasks to keep the application up-to-date.

### CODE REVIEW

File: apps/business_central/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    Error.objects.update_or_create(workspace_id=accounting_export.workspace_id, accounting_export=accounting_export, defaults={'error_title': error_msg, 'type': 'BUSINESS_CENTRAL_ERROR', 'error_detail': business_central_error, 'is_resolved': False})
+
+    accounting_export.status = 'FAILED'
+    accounting_export.detail = None
+    accounting_export.business_central_errors = business_central_error
+    accounting_export.save()
+
+
+def set_last_export_details(accounting_export: AccountingExport, status, message):
```
Comment (NileshPant1999) [inline apps/business_central/exceptions.py:30]: where is this function used ??

File: apps/business_central/exports/journal_entry/queues.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                'status': 'ENQUEUED',
+                'type': 'JOURNAL_ENTRY'
+            }
+        )
+
+        if accounting_export.status not in ['IN_PROGRESS', 'ENQUEUED']:
+            accounting_export.status = 'ENQUEUED'
+            accounting_export.save()
+
+        """
```
Comment (NileshPant1999) [inline apps/business_central/exports/journal_entry/queues.py:38]: please fix this,
this is done in sage300

**Review by NileshPant1999 — APPROVED** [review]

File: apps/business_central/exports/journal_entry/queues.py (hunk 2 above)
Comment (ruuushhh) [reply apps/business_central/exports/journal_entry/queues.py:38]: will fix

File: apps/business_central/exceptions.py (hunk 1 above)
Comment (ruuushhh) [reply apps/business_central/exceptions.py:30]: will remove this

**Pull request merged**